[One Workflow] Add timezone suggestions for tzid field in scheduled triggers

VladimirFilonov opened the pull request:
## Summary

closes: https://github.com/elastic/security-team/issues/14189

<img width="680" height="546" alt="image" src="https://github.com/user-attachments/assets/65b6898d-07a7-43c0-ac08-5c0e132cbf8b" />

VladimirFilonov committed:
Add timezone suggestions for tzid field in scheduled triggers

semd reviewed (commented):
Looking good, but let's make sure the approach is performant

VladimirFilonov committed:
Merge remote-tracking branch 'upstream/main' into feature/14189-show-timezone-autocompletion-suggestions

VladimirFilonov committed:
- Introduced a constant for sorted timezone names to improve performance.
- Updated the timezone suggestion logic to handle both 'tzid' and 'timezone' fields.
- Reduced the limit of timezone suggestions from 50 to 25 for better performance.

semd reviewed (approved):
LGTM! 
Thanks for addressing the comments! 💯 

As a follow-up, we could use `const userTz = moment.tz.guess();` to pre-set the current user TZ, or show it first in the suggestions list.
